### PR TITLE
Add ResxSourceGenerator

### PR DESCRIPTION
**Pull request by sharwell**
This is generally extracted from https://github.com/dotnet/arcade/blob/master/src/Microsoft.DotNet.Arcade.Sdk/src/GenerateResxSource.cs.

### CODE REVIEW

File: samples/CSharp/SourceGenerators/SourceGeneratorSamples/ResxSourceGenerator.cs [hunk 1]
```diff
@@ -0,0 +1,663 @@
...
+                return;
+            }
+
+            if (!context.AnalyzerConfigOptions.GlobalOptions.TryGetValue("build_property.RootNamespace", out string rootNamespace))
+            {
+                rootNamespace = context.Compilation.AssemblyName;
+            }
+
+            string resourceName = Path.GetFileNameWithoutExtension(resourceFile.Path);
+            if (options.TryGetValue("build_metadata.AdditionalFiles.RelativeDir", out string relativeDir))
```
Comment (sharwell) [inline samples/CSharp/SourceGenerators/SourceGeneratorSamples/ResxSourceGenerator.cs:58]: @chsienki This value is appearing in the generated global **.editorconfig** file, but it's not appearing in `options`

Comment (chsienki) [reply samples/CSharp/SourceGenerators/SourceGeneratorSamples/ResxSourceGenerator.cs:58]: This was fixed in https://github.com/dotnet/roslyn/pull/45357 but hasn't flowed through to any published packages yet.

Comment (sharwell) [reply samples/CSharp/SourceGenerators/SourceGeneratorSamples/ResxSourceGenerator.cs:61]: ➡️ Fixed by #559

**Comment by sharwell** [comment]
Superseded by https://github.com/dotnet/roslyn-analyzers/pull/4897

**Pull request closed without merging**